### PR TITLE
Feat: Adds `RegionModal` component

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

`fs/components`
- [x] Renames and moves `RegionModal` component
- [x] Adds `RegionModal ` doc page

`fs/ui`
- [x] Creates `RegionModal` styles

`fs/core`
- [x] Review `RegionModal` structure (Removes `RegionInput`)
- [x] Delete  old `Region`'s  related files (storybook and style)

## How to test it?

run `yarn dev` on `core` application and check the `RegionModal` feature on the homepage (Click on the `RegionBar` - Set your location button) or test it using the doc preview link:
- [RegionModal](https://faststore-site-git-feat-region-modal-fs-810-faststore.vercel.app/components/organisms/region-modal)

### Starters Deploy Preview

[WIP]

### CODE REVIEW

**Review by renatamottam — APPROVED** [review]
Nice work, fan! Left a few suggestions :)

File: packages/components/src/organisms/RegionModal/RegionModal.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+              onClose?.()
+            }}
+            title={title}
+            description={description}
+            closeBtnProps={{
+              'aria-label': 'Close Region Modal',
+            }}
+          />
+          <ModalBody>
+            <InputField
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply packages/components/src/organisms/RegionModal/RegionModal.tsx:94]: @renatamottam Thanks for reviewing :)

Is this an [InputField](https://evergreen.faststore.dev/components/molecules/input-field) default behavior or just for this input inside the `RegionModal`? 

This input behavior should be controlled by the component that is calling it.

**Review by eduardoformiga — APPROVED** [review]
good Job!

**Review by ArthurTriis1 — APPROVED** [review]

File: packages/components/src/organisms/RegionModal/RegionModal.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (hellofanny) [reply packages/components/src/organisms/RegionModal/RegionModal.tsx:94]: I think it's something that should be controlled by the component above.

**Pull request merged**